### DISCUSSION TITLE
PrismaClient is unable to be run in the browser.

### DISCUSSION THREAD

**Discussion by weiwei2694** [prisma/prisma#20144] [Q&A] [answered]
I have a little problem with Prisma, I'm using NextJs, no typescript. The code I have is below:

```js
import { PrismaClient } from "@prisma/client";

let prisma;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;
```

This code previously worked, and there were no problems, during my journey working on this project, suddenly this code could not be reused

**Comment by ludralph** [comment]
Hi @weiwei2694, Thank you for raising this question. 

How are you using the exported `prisma` instance in your Next.js app? 

If you are trying to use Prisma on frontend, it is not possible. Either use it in Next js api routes: https://nextjs.org/docs/api-routes/introduction or use it in getServerSideProps or getStaticProps: https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering. If you are on Next.js 13, API Routes have been replaced by Route Handlers in Next.js 13.2

**Reply by weiwei2694** [reply]
is there any problem using it in server component?

**Reply by weiwei2694** [reply]
I use it in the api, when working in the client component, and when I work in the server component, I immediately use prisma, without api at all

**Reply by ludralph** [reply]
Can you please share a minimal reproduction so I can take a look?

**Reply by weiwei2694** [reply, marked as answer]
hi, sorry for just replying to your message, I've solved the problem, the error is in NextJS itself, I just changed my Code Prisma to:
```
import { PrismaClient } from '@prisma/client';
const prisma = new PrismaClient()

export default prisma`

after that, I turned off developer mode('npm run dev), then turned it on again, and it worked, then I changed the code again to before:
`import { PrismaClient } from "@prisma/client";

let prisma;

if (process.env.NODE_ENV === "production") {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}

export default prisma;
```

weird, but it's the truth. Thanks for helping in advance
